code/bmpman/bmpman.cpp

```diff
@@ -1663,6 +1773,58 @@ void bm_lock_ani(int handle, int bitmapnum, bitmap_entry *be, bitmap *bmp, ubyte
 	anim_free(the_anim);
 }
 
+
+void bm_lock_apng(int handle, int bitmapnum, bitmap_entry *be, bitmap *bmp, ubyte bpp, ubyte flags) {
+
+	int first_frame = be->info.ani.first_frame;
+	int nframes = bm_bitmaps[first_frame].info.ani.num_frames;
+
+	apng::apng_ani* the_apng;
```

Comment:
You can use `.reset()` to assign a new pointer value to a unique_ptr. No need to create a temporary pointer instance.